src/app/task/task.component.ts

```diff
@@ -13,5 +15,47 @@ import { MatToolbarModule } from '@angular/material/toolbar';
   templateUrl: './task.component.html',
   styleUrl: './task.component.scss'
 })
-export class TaskComponent {
+export class TaskComponent implements OnInit, OnDestroy {
+  private destroy$: Subject<void> = new Subject<void>();
+
+  authService = inject(AuthService);
+
+  constructor (private router: Router) {}
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  logout() {
+    const refreshToken = localStorage.getItem('refreshToken');
+
+    if (refreshToken) {
+      this.authService.logout(refreshToken)
```

Comment:
Do you need a token for login/logout? they're not really protected endpoints are they?

Reply:
and if you do use it, why do you use refreshToken instead of accessToken?